include/class-monero-gateway.php

```diff
@@ -209,12 +209,24 @@ public function process_payment($order_id)
 
         $order = wc_get_order($order_id);
 
-        // Generate a unique payment id
-        do {
-            $payment_id = bin2hex(openssl_random_pseudo_bytes(8));
-            $query = $wpdb->prepare("SELECT COUNT(*) FROM $table_name WHERE payment_id=%s", array($payment_id));
-            $payment_id_used = $wpdb->get_var($query);
-        } while ($payment_id_used);
+        if(self::$confirm_type != 'monero-wallet-rpc') {
+          // Generate a unique payment id
+          do {
+              $payment_id = bin2hex(openssl_random_pseudo_bytes(8));
+              $query = $wpdb->prepare("SELECT COUNT(*) FROM $table_name WHERE payment_id=%s", array($payment_id));
+              $payment_id_used = $wpdb->get_var($query);
+          } while ($payment_id_used);
+        }
+        else {
+          // Generate subaddress
+          $payment_id = self::$monero_wallet_rpc->create_address(0, 'Order: ' . $order_id);
```

Comment:
* Use "auto-describing" name for variables (i.e payment_id => subaddress_rpc ).

Reply:
I agree of course, but as this is a half upgrade (until math is converted to php and the non-rpc version uses subaddresses also) so not using the $payment_id variable means I would need to have a lot of extra code to make sure $subaddress is being used instead of $payment_id, another sanitizing function, another db table column and checks all along the way. It would be easier to do the full upgrade than balancing both methods with separate variables. I liked the simplicity of just using that variable. I'm not saying I'm right though; it's not a big deal to put those checks in I suppose.

Today I'm testing if the explorer will be suitable for subaddresses like it is payment ids. If it is I'll ask r/monero to help convert knacc's subaddress javascript code to php and do the full upgrade instead.